### PR TITLE
Implement announcement channel following

### PR DESCRIPTION
**Pull request by GnomedDev**
Quite surprised this wasn't implemented. I'm not the best doc writer and I'm probably inconsistent.
https://discord.com/developers/docs/resources/channel#follow-news-channel

### CODE REVIEW

File: src/http/client.rs [hunk 1]
```diff
@@ -1765,6 +1765,26 @@ impl Http {
         .await
     }
 
+    /// Follow an Announcement Channel to send messages to a target channel.
+    pub async fn follow_news_channel(
```
Comment (Milo123459) [inline src/http/client.rs:1769]: Should this be named `follow_news_channel` or `follow_announcement_channel`?

Comment (GnomedDev) [reply src/http/client.rs:1769]: The API calls them news channels, but I saw existing references to announcement channels in serenity. I thought for the raw http module I would follow the api name then could just call it `follow` on Channel(Id) and avoid the naming bikeshedding.

Comment (Milo123459) [reply src/http/client.rs:1769]: Maybe we should rename the existing references to news channels?

Comment (GnomedDev) [reply src/http/client.rs:1769]: Should this been done in a different PR to next since that would be breaking?

Comment (arqunis) [reply src/http/client.rs:1769]: We should strive to be consistent as possible, so yes, a pull request targeting `next` should be opened to replace instances of "announcement channel" with "news channel".

Comment (Milo123459) [reply src/http/client.rs:1769]: Why not just target this PR at next and replace all references in this PR?

Comment (kangalio) [reply src/http/client.rs:1769]: Probably best to keep this PR like it is and maybe make a follow up PR

Small PRs are good anyways

**Review by arqunis — APPROVED** [review]

**Pull request merged**